pallets/asset-manager/src/lib.rs

```diff
@@ -394,6 +396,78 @@ pub mod pallet {
 		pub fn account_id() -> T::AccountId {
 			T::PalletId::get().into_account()
 		}
+
+		pub fn set_genesis_configuration() -> Weight {
+			let mut weight: Weight = T::DbWeight::get().reads(1);
+			if have_storage_value(Self::name().as_bytes(), b"NextAssetId", &[]) {
+				log::warn!(" !!! Aborting, NextAssetId was already set.");
+				return weight;
+			}
+
+			weight = weight.saturating_add(T::DbWeight::get().reads(4));
+			let asset_id = <T::AssetConfig as AssetConfig<T>>::NativeAssetId::get();
+			let metadata = <T::AssetConfig as AssetConfig<T>>::NativeAssetMetadata::get();
+			let location = <T::AssetConfig as AssetConfig<T>>::NativeAssetLocation::get();
+			if have_storage_value(
+				Self::name().as_bytes(),
+				b"AssetIdLocation",
+				&Blake2_128Concat::hash(&asset_id.encode()),
+			) {
+				log::warn!(" !!! Aborting, AssetIdLocation was already set.");
+				return weight;
```

Comment:
If this branch happens, the rest of code will not be executed. Is it expected?

Reply:
Yes, we haven't done any writes so far so we can exit early.